### PR TITLE
Adding support for assigning workspace capacity

### PR DESCRIPTION
**Pull request by ladlavj**
Assigns capacity ID to workspace

### CODE REVIEW

**Comment by ladlavj** [comment]
Secrets from my repository are being ignored I believe it's mentioned here and not possible from forked repository.
https://docs.github.com/en/free-pro-team@latest/actions/reference/encrypted-secrets#using-encrypted-secrets-in-a-workflow
"With the exception of GITHUB_TOKEN, secrets are not passed to the runner when a workflow is triggered from a forked repository."

do you have any other idea?

File: docs/resources/workspace.md [hunk 1]
```diff
@@ -14,11 +14,12 @@ resource "powerbi_workspace" "myworkspace" {
 #### The following arguments are supported:
 <!-- docgen:NonComputedParameters -->
 * `name` - (Required, Forces new resource) Name of the workspace.
+* `capacity_id` - (Optional) Capacity ID to be assigned to workspace.
```
Comment (alex-davies) [inline docs/resources/workspace.md:17]: Would be good to indicate what prerequisites and permissions are required to be able to use capacity_id

**Comment by alex-davies** [comment]
> Secrets from my repository are being ignored I believe it's mentioned here and not possible from forked repository.
> https://docs.github.com/en/free-pro-team@latest/actions/reference/encrypted-secrets#using-encrypted-secrets-in-a-workflow
> "With the exception of GITHUB_TOKEN, secrets are not passed to the runner when a workflow is triggered from a forked repository."
> 
> do you have any other idea?

Thats annoying, seems like github dont even have any possible solutions. Best I can do is merge it into a branch on this repo then merge that branch into master.

**Comment by alex-davies** [comment]
There are no tests for this change, although even if there were I wont be able to run them as I don't have a premium licence. Do you have a premium licence which you have tested this against?

Is it possible to introduce some test cases and a POWERBI_IS_PREMIUM flag to only run the tests if the licence is a premium licence?

**Comment by ladlavj** [comment]
> There are no tests for this change, although even if there were I wont be able to run them as I don't have a premium licence. Do you have a premium licence which you have tested this against?
> 
> Is it possible to introduce some test cases and a POWERBI_IS_PREMIUM flag to only run the tests if the licence is a premium licence?

Yes..feature does work, I've tested in my environment. I will try to implement POWERBI_IS_PREMIUM flag

File: docs/resources/workspace.md (hunk 1 above)
Comment (ladlavj) [reply docs/resources/workspace.md:20]: Updated the workspace.md with the minimum required permissions.

**Comment by ladlavj** [comment]
Added PowerBI workspace capacity assigning tests, and controlling flag `POWERBI_IS_PREMIUM` can be set to `true` or `false`
Tests would be included if `POWERBI_IS_PREMIUM` is set to `true` in addition to `POWERBI_CAPACITY_ID`.

**Pull request merged**